tests/drivers/can/api/src/common.c

```diff
@@ -244,3 +244,13 @@ void can_common_test_setup(can_mode_t initial_mode)
 	err = can_start(can_dev);
 	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
 }
+
+void before_test(void *not_used)
```

Comment:
This is improvement.
Test cases assume that CAN is started when test begins.
There are test that stop CAN, do testing, re-start CAN.
If such test fails on assertion then test execution is terminated. As a result, CAN is not restarted and precondition (CAN started) is not meet for remaining tests.
Here, I add code that is executed before each test. I ignore error code
-EALREADY	if the device is already started.
https://docs.zephyrproject.org/latest/doxygen/html/group__can__interface.html#ga343456749eec6144a91bacae0d94b114

Reply:
Please remove this part. This is not sufficient for restoring the CAN controller to a default status in all failure scenarios (e.g. installed filters, but test case failed) and as such provide limited value.

If needed, we can move this to a separate PR/issue.

Reply:
"This is not sufficient for restoring the CAN controller to a default state (...)"
You are right, "test case setup function" doesn't restore other CAN parameters such as installed filters, bitrate, timing, etc.
Removing.